### ISSUE TITLE
ERROR Module build failed: TypeError: ruleConfiguration.forEach is not a function

### ISSUE THREAD

**Issue by duffbuster** [wbuchwalter/tslint-loader#73]
Getting an error when running Webpack.
```
ERROR in ./src/index.ts
Module build failed: TypeError: ruleConfiguration.forEach is not a function
    at Object.convertRuleOptions (/Users/colin/Projects/CMS-Web/node_modules/tslint/lib/configuration.js:357:23)
    at Linter.getEnabledRules (/Users/colin/Projects/CMS-Web/node_modules/tslint/lib/linter.js:194:47)
    at Linter.lint (/Users/colin/Projects/CMS-Web/node_modules/tslint/lib/linter.js:76:33)
    at lint (/Users/colin/Projects/CMS-Web/node_modules/tslint-loader/index.js:58:10)
    at Object.module.exports (/Users/colin/Projects/CMS-Web/node_modules/tslint-loader/index.js:128:3)
```

My Webpack config us just
```
{
        test: /\.ts$/,
        exclude: /node_modules/,
        loader: 'tslint-loader',
        enforce: 'pre'
},
```

tslint version: 5.1.0
tslint-loader version: 3.5.2
Webpack version: 2.4.1

**Comment by mattiLeBlanc** [comment]
+1

having the same issue.
I moved the tslint configuration line into the rule:
`{
        enforce: 'pre',
        test: /.ts$/,
        loader: 'tslint-loader',
        exclude: /node_modules/,
        options: {
          configuration:  require( '../tslint.json' )
        }
      },`
Still same issue.
I dumped the TSLint json and it looks good. There are no errors in my lint configuration.

**Comment by mattiLeBlanc** [comment]
Okay, started to debug and I found the following:
The rule configuration going into the `convertRuleOptions` function in `tslint/lib/configuration: line 355` is an object literal, and you are trying to do a forEach on an object literal which is not a native function of Object.
So either use the Object.keys() and iterate over the keys or add Lodash forEach to iterate over this object.
I am happy to make a pull request?

**Comment by mattiLeBlanc** [comment]
Or maybe you want to look at my solution first, just VanillaJS:
```
function convertRuleOptions(ruleConfiguration) {

    var output = [];
    var ruleKeys = Object.keys(ruleConfiguration);
    ruleKeys.forEach( function( key ) {
        var partialOptions = ruleConfiguration[ key ];
        var ruleName = key;
        var options = {
            disabledIntervals: [],
            ruleArguments: partialOptions.ruleArguments || [],
            ruleName: ruleName,
            ruleSeverity: partialOptions.ruleSeverity || "error",
        };
        output.push(options);
    });
    // console.log(output);
    // ruleConfiguration.forEach(function (partialOptions, ruleName) {
    //     var options = {
    //         disabledIntervals: [],
    //         ruleArguments: partialOptions.ruleArguments || [],
    //         ruleName: ruleName,
    //         ruleSeverity: partialOptions.ruleSeverity || "error",
    //     };
    //     output.push(options);
    // });
    return output;
}
```

**Comment by blacksonic** [comment]
The raw configuration object needs to be transformed in a way that `findConfigurationFile` works.

**Comment by mattiLeBlanc** [comment]
Can you show me how? I am just using the tslint-loader, upgraded to a newer version of tslint, tslint-loader and webpack. And since that upgrade I got this issue.

**Comment by blacksonic** [comment]
For Tslint 5 the raw configuration needs to be transformed with `parseConfigFile`. Fixed in version 3.5.3

**Comment by mattiLeBlanc** [comment]
I can confirm it solves the problem. Thank you.

**Comment by DenisVuyka** [comment]
The issue is not fixed with 3.5.3 upgrade

**Comment by Izhaki** [comment]
@blacksonic comment was immensely helpful.

Just in case it helps anyone else, there's an example of working usage of `parseConfigFile` [here](https://github.com/palantir/tslint/issues/2457#issuecomment-332822605).
